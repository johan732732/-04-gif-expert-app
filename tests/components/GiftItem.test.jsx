const { render, screen } = require('@testing-library/react');
const { GifItem } = require('../../src/components/GifItem');

describe('Test in <GiftItem/>', () => {
  const title = 'Hola, soy goku';
  const url = 'http://dragon-ball/goku.jpg';

  test('Must match with the snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Must show the image with the title', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Must to show the tile in the component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
