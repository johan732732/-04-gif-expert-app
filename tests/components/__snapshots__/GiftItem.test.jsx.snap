// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GiftItem/> Must match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola, soy goku"
      src="http://dragon-ball/goku.jpg"
    />
    <p>
      Hola, soy goku
    </p>
  </div>
</div>
`;
